api/ask_astro/services/questions.py

```diff
@@ -27,6 +28,20 @@ async def _update_firestore_request(request: AskAstroRequest) -> None:
     )
 
 
+def _preprocess_request(request: AskAstroRequest) -> None:
+    if len(request.prompt) > 20000:
+        error_msg = "Question text is too long. Please try making a new thread and shortening your question."
+        request.response = error_msg
+        raise Exception(error_msg)
+    if not request.prompt:
+        error_msg = "Question text cannot be empty. Please try again with a different question."
+        request.response = error_msg
+        raise Exception(error_msg)
```

Comment:
Could we use a custom exception instead of using Execption here?

Reply:
Oops my bad I forgot to push my changes haha. I was like I thought I fixed this. Please feel free to re-review and resolve this comment if it's good!